foundation-ui/auth/AuthService.js

```diff
@@ -0,0 +1,67 @@
+import {EventEmitter} from 'events';
+
+import Authorizer from './Authorizer';
+import {CHANGE} from './AuthEvent';
+
+/**
+ * List of authorizers
+ *
+ * @type {Array.<Authorizer>}
+ */
+const authorizers = [];
+
+/**
+ * AuthService
+ */
+class AuthService extends EventEmitter {
+
+  /**
+   * Register authorizer
+   *
+   * @param {Authorizer} authorizer
+   */
+  registerAuthorizer(authorizer) {
+    if (!authorizer || !(authorizer instanceof Authorizer)) {
```

Comment:
The `authorize` method doesn't have to return a boolean, just like a regular function, it could return anything too. I'm ok with the proposal - just wanted to understand the extra layer required of extending from an existing class :)